gnd/src/main/java/com/google/android/gnd/ui/editrecord/EditRecordViewModel.java

```diff
@@ -88,17 +89,31 @@
     this.argumentProcessor = BehaviorProcessor.create();
     this.recordProcessor = BehaviorProcessor.create();
 
-    Flowable<Resource<Record>> recordStream =
+    Flowable<Resource<Record>> saveRecordStream =
         recordProcessor.flatMap(
             r ->
                 this.authManager
                     .getUser()
                     .flatMap(user -> this.dataRepository.saveChanges(r, getChangeList(r), user))
                     .toFlowable(BackpressureStrategy.LATEST));
 
-    Single<Resource<Record>> newRecord = argumentProcessor.map()
-
-    disposeOnClear(recordStream.subscribe(record::setValue));
+    Flowable<Resource<Record>> editRecordStream =
```

Comment:
I think there might be a better approach here, but this was the easiest to implement and straightforward. 

There's a distinction in RX between operating on the emitted items of the stream (maps) and operating on the entire stream itself (compose) using stream transformers. I have a hunch there may be some cases in which using these sorts of transformers could help simplify cases like this, where we manipulate some source stream to create N additional streams using transformations.

Reply:
I don't know if we necessarily need to use custom transformers here, maybe breaking some of the logic into helper methods here and elsewhere would make it more readable?